readiness/e2e/suite_test.go

```diff
@@ -0,0 +1,470 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1alpha2 "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha2"
+)
+
+var (
+	timeout         = 2 * time.Minute
+	pollingInterval = 50 * time.Millisecond
+	testNamespace   = "default"
+)
+
+var _ = Describe("Readiness end to end suite", func() {
+	Context("Readiness with no checks", func() {
+		err := cl.Create(context.TODO(), &v1alpha2.Readiness{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "readiness-without-checks",
+			},
+			Spec: v1alpha2.ReadinessSpec{
+				Checks: []v1alpha2.Check{},
+			},
+		})
+		Expect(err).To(BeNil())
+
+		It("Should reconcile to ready state", func() {
+			Eventually(func() bool {
+				readiness := &v1alpha2.Readiness{}
+				err := cl.Get(context.TODO(), types.NamespacedName{Name: "readiness-without-checks"}, readiness)
+				return err == nil && readiness.Status.Ready == true
+			}).WithTimeout(5 * timeout).WithPolling(pollingInterval).Should(BeTrue())
+
+			deleteReadinessResources()
+		})
+	})
+
+	Context("Readiness with one check and one provider", func() {
+		It("Should be reconciled to non-ready state", func() {
+			err := cl.Create(context.TODO(), &v1alpha2.Readiness{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "readiness-with-one-check-1",
+				},
+				Spec: v1alpha2.ReadinessSpec{
+					Checks: []v1alpha2.Check{
+						{
+							Name:     "check1",
+							Type:     "basic",
+							Category: "test",
+						},
+					},
+				},
+			})
+			Expect(err).To(BeNil())
+
+			Eventually(func() bool {
+				readiness := &v1alpha2.Readiness{}
+				err := cl.Get(context.TODO(), types.NamespacedName{Name: "readiness-with-one-check-1"}, readiness)
+				return err == nil && readiness.Status.Ready == false
+			}).WithTimeout(timeout).WithPolling(pollingInterval).Should(BeTrue())
+		})
+
+		It("Should be reconciled to ready state", func() {
+			Eventually(func() bool {
+				err := cl.Create(context.TODO(), &v1alpha2.ReadinessProvider{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "check-1-provider",
+					},
+					Spec: v1alpha2.ReadinessProviderSpec{
+						CheckRefs:  []string{"check1"},
+						Conditions: []v1alpha2.ReadinessProviderCondition{},
+					},
+				})
+				return err == nil
+			}).WithTimeout(timeout).WithPolling(pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				readiness := &v1alpha2.Readiness{}
+				err := cl.Get(context.TODO(), types.NamespacedName{Name: "readiness-with-one-check-1"}, readiness)
+				return err == nil && readiness.Status.Ready == true
+			}).WithTimeout(timeout).WithPolling(pollingInterval).Should(BeTrue())
+
+			deleteReadinessResources()
+		})
+	})
+
+	Context("Readiness with one check and two providers", func() {
+		It("Should create readiness object successfully", func() {
```

Comment:
`Should create Readiness and ReadinessProvider objects successfully`

Reply:
Made some changes to the structure of the code. Please let me know if this is still valid.

Reply:
Made some changes to the structure of the code. Please let me know if this is still valid.